Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
```

Comment:
Please move PyObject_GC_UnTrack after this if block, and remove Track and Untrack in the block.